sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/keyvault_exception.hpp

```diff
@@ -11,35 +11,24 @@
 #include <azure/core/exception.hpp>
 #include <azure/core/http/http.hpp>
 
+#include <memory>
 #include <stdexcept>
 #include <string>
 
-namespace Azure { namespace Security { namespace KeyVault {
-
+namespace Azure { namespace Security { namespace KeyVault { namespace _detail {
   /**
-   * @brief The general exception thrown by the Key Vault SDK clients.
+   * @brief Container for static methods to parse keyvault payloads to Azure Core Exception.
    *
    */
-  class KeyVaultException : public Azure::Core::RequestFailedException {
-  public:
-    /**
-     * @brief Construct a new Key Vault Exception object without an Http raw response.
-     *
-     * @remark A Key Vault Exception without an Http raw response represent an exception happend
-     * before sending the request to the server. There is no response yet.
-     *
-     * @param message An error message for the exception.
-     */
-    explicit KeyVaultException(const std::string& message) : RequestFailedException(message) {}
-
+  struct KeyVaultException
```

Comment:
It belongs to the Service because only the service knows about the service-response-payload.

For all the common-data across services (like http code, reason-phrase, etc), Azure-core can expect to get those fields from a raw-response from any service response.

But, for things like message or error code (or any other field), the data might be in a header, or in the XML/Json/text payload.  Only the service knows about hot to get that data.

That's way I remove the type but let the logic for getting the data from the raw-response and passing it to Azure-Core exception.